Azure-Samples/chat-with-your-data-solution-accelerator
# Question Answer Tool changes

## Purpose

* Related to #321 
* This PR updates QuestionAnswerTool.py to work with Integrated Vectorization. It adds a new field 'chunk_id' to the lang chain SourceDocument.py class to be able to map it in UI.
* Added corresponding unit tests

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
The Web chat application should be able to return results along with citations for the ingested documents.
```
